gui-js/libs/core/src/lib/services/communication/communication.service.ts

```diff
@@ -393,8 +394,16 @@ export class CommunicationService {
       case 'mousedown':
         if (message.ctrlKey || message.metaKey)
           canvas.controlMouseDown(this.mouseX, this.mouseY);
-        else
+        else {
+          this.electronService.send(events.CONTEXT_MENU, {
```

Comment:
this where we ALWAYS propagate the context menu event upon left click. Care should be taken that the logic exits early if no context menu showing is required, as this will happen in more than 99% of cases...